src/main/java/org/folio/circulation/domain/policy/LoanPolicyRepository.java

```diff
@@ -34,6 +41,45 @@ public LoanPolicyRepository(Clients clients) {
       .thenApply(result -> result.map(relatedRecords::withLoanPolicy));
   }
 
+  public CompletableFuture<Result<Loan>> findPolicyForLoan(Result<Loan> loanResult) {
+    return loanResult.after(loan ->
+      getLoanPolicyByID(loan.getLoanPolicyId())
+      .thenApply(result -> result.map(loan::withLoanPolicy)));
+  }
+
+  private CompletableFuture<Result<LoanPolicy>> getLoanPolicyByID(String loanPolicyId) {
+    return FetchSingleRecord.<LoanPolicy>forRecord("loansPolicy")
+            .using(loanPoliciesClient)
+            .mapTo(LoanPolicy::from)
+            .whenNotFound(succeeded(null))
+            .fetch(loanPolicyId);
+  }
+
+  public CompletableFuture<Result<MultipleRecords<Loan>>> findLoanPoliciesForLoans(MultipleRecords<Loan> multipleLoans) {
+
+    Collection<Loan> loans = multipleLoans.getRecords();
+
+    return getLoanPolicies(loans).thenApply(r -> r
+      .map(loanPolicies -> multipleLoans.mapRecords(loan -> loan.withLoanPolicy(loanPolicies.getOrDefault(loan.getLoanPolicyId(), null)))));
```

Comment:
Please split this line so that it fits within the intended 80 character line length

Reply:
Done